spec/unit/util/run_mode_spec.rb

```diff
@@ -2,6 +2,13 @@
 require 'spec_helper'
 
 describe Puppet::Util::RunMode do
+
+  # Discriminator for tests that attempts to unset HOME since that, for reasons currently unknown,
```

Comment:
I'm not seeing a problem (with core Ruby `ENV`), at least on OSX / Ruby 2.4.0

```
irb(main):001:0> RUBY_VERSION
=> "2.4.0"
irb(main):002:0> ENV['HOME']
=> "/Users/Iristyle"
irb(main):003:0> ENV['HOME'] = nil
=> nil
irb(main):004:0> ENV['HOME']
=> nil

irb(main):009:0> ENV.keys.include?('HOME')
=> false
```

With our helpers:

```
[1] pry(main)> require 'puppet'
=> true
[2] pry(main)> Puppet::Util.get_env('HOME')
=> "/Users/Iristyle"
[3] pry(main)> Puppet::Util.set_env('HOME')
=> nil
[4] pry(main)> Puppet::Util.get_env('HOME')
=> nil
[5] pry(main)> RUBY_VERSION
=> "2.4.0"
[6] pry(main)> Puppet.version
=> "5.0.0"
```